### PR TITLE
[EVM] Remove special EVM type

### PR DESCRIPTION
**Pull request by sideninja**
Closes: #5414 

**💔 API Breaking Change**

Remove the special event type for evm: `evm.BlockExecuted` and `evm.TransactionExecuted` and make them be generic Flow event types: `A.{evm contract address}.EVM.TransactionExecuted` and `A.{evm contract address}.EVM.BlockExecuted`.

### CODE REVIEW

**Review by janezpodhostnik — APPROVED** [review]

**Review by ramtinms — APPROVED** [review]

**Comment by turbolent** [comment]
Could you also please remove the EVM location again from the Go SDK?

File: fvm/evm/evm_test.go [hunk 1]
```diff
@@ -97,15 +97,49 @@ func TestEVMRun(t *testing.T) {
...
+				// assert event fiedls are correct
+				require.Len(t, output.Events, 2)
+
+				blockEvent := output.Events[1]
+
+				assert.Equal(t, fmt.Sprintf(
+					"A.%s.%s",
+					sc.EVMContract.Address.String(),
+					types.EventTypeBlockExecuted,
+				), string(blockEvent.Type))
```
Comment (turbolent) [inline fvm/evm/evm_test.go:109]: Use `common.Location.TypeID()` instead. Same below

File: fvm/fvm_test.go [hunk 2]
```diff
@@ -3278,10 +3278,9 @@ func TestEVM(t *testing.T) {
 			require.NoError(t, output.Err)
 			require.Len(t, output.Events, 7)
 
-			evmLocation := types.EVMLocation{}
 			txExe, blockExe := output.Events[4], output.Events[5]
-			assert.Equal(t, evmLocation.TypeID(nil, string(types.EventTypeTransactionExecuted)), common.TypeID(txExe.Type))
-			assert.Equal(t, evmLocation.TypeID(nil, string(types.EventTypeBlockExecuted)), common.TypeID(blockExe.Type))
+			assert.Equal(t, fmt.Sprintf("A.%s.%s", sc.EVMContract.Address, types.EventTypeTransactionExecuted), string(txExe.Type))
+			assert.Equal(t, fmt.Sprintf("A.%s.%s", sc.EVMContract.Address, types.EventTypeBlockExecuted), string(blockExe.Type))
 		}),
```
Comment (turbolent) [inline fvm/fvm_test.go:3284]: Keep using `common.Location.TypeID` here

**Comment by sideninja** [comment]
> Could you also please remove the EVM location again from the Go SDK?

@m-Peter plans to

File: fvm/fvm_test.go (hunk 2 above)
Comment (sideninja) [reply fvm/fvm_test.go:3284]: fixed

**Comment by turbolent** [comment]
@sideninja Maybe forgot to push? There are no new commits

**Review by m-Peter — CHANGES_REQUESTED** [review]
@sideninja There are a couple more files that ideally should be updated:
- https://github.com/onflow/flow-go/blob/master/model/flow/event.go#L18
- https://github.com/onflow/flow-go/blob/master/model/events/parse.go#L28-L35
- https://github.com/onflow/flow-go/blob/master/model/events/parse_test.go#L46-L57

It would also be very useful to address this suggestion as well: https://github.com/onflow/flow-go/issues/5414#issuecomment-2071573662

We should also include the event definitions in the `EVM` contract itself, for example:

```cadence
access(all)
event BlockExecuted(
    height: UInt64,
    hash: String,
    ...
)

access(all)
event TransactionExecuted(
    blockHeight: UInt64,
    blockHash: String,
    hash: String,
    payload: String,
    ...
    logs: String
)
```

This will be helpful for other tools (such as the Cadence testing framework), to make use of these types cc @turbolent

**Comment by sideninja** [comment]
> @sideninja Maybe forgot to push? There are no new commits

I did, sorry

**Comment by sideninja** [comment]
> @sideninja There are a couple more files that ideally should be updated:
> 
> * https://github.com/onflow/flow-go/blob/master/model/flow/event.go#L18
> * https://github.com/onflow/flow-go/blob/master/model/events/parse.go#L28-L35
> * https://github.com/onflow/flow-go/blob/master/model/events/parse_test.go#L46-L57
> 
> It would also be very useful to address this suggestion as well: [#5414 (comment)](https://github.com/onflow/flow-go/issues/5414#issuecomment-2071573662)
> 
> We should also include the event definitions in the `EVM` contract itself, for example:
> 
> ```cadence
> access(all)
> event BlockExecuted(
>     height: UInt64,
>     hash: String,
>     ...
> )
> 
> access(all)
> event TransactionExecuted(
>     blockHeight: UInt64,
>     blockHash: String,
>     hash: String,
>     payload: String,
>     ...
>     logs: String
> )
> ```
> 
> This will be helpful for other tools (such as the Cadence testing framework), to make use of these types cc @turbolent

Great feedback, thanks. I like the idea of defining the event in the contract.

**Review by m-Peter — APPROVED** [review]
LGTM! :ok_hand:

**Review by janezpodhostnik — APPROVED** [review]

**Review by turbolent — APPROVED** [review]
Nice!

**Pull request merged**